### ISSUE TITLE
HTML formatting: even with --html-whitespace-sensitivity strict formatter still adding new lines inside the string

### ISSUE THREAD

**Issue by alexdinescu** [prettier/prettier#8338]
**Prettier 2.0.5**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4AIwA2AhgM4kC8AOiEfsWdbpgIwAcUAfARAE5wC2TAA4kAroPQRuPXCUwxcRfnBjSBwsRKm9Z8xctW91mEeNyTpuhUpVrBJzee0y51g3Y1mLO1-pUA6VAB6LGwOSigMHDpSCmpaQljGFnYuI3tTLUtfG0M+DMdvGWDQjhAAGhAIIRhMaBJkUCIeHggAdwAFZoQGlCICNqIATwbKgCMeIjAAaxUAZSEpzCgAc2QYHlE4SoALGH4CAHUd+TgSRbA4OZ75HHkh5BASDcwwGAqQZZI4HhgOyZW-CIyAAZv1vpUAFYkAAeACFJjN5jYADLLOCg8HbEDQmFzZYrAhwACKogg8ExBAhIEWPG+PEeewOHyEPGWMEOmHQMB2yFYAAZKqyIN9DpMhI9WWcftgMZUAI5k+D-aq9GgkAC0UDgcHQuo+fEVmD4-yIgOBSDBVOx334mHWmxtBKJpPJGKQGy2lRgRDGnO5vKQACZvZNMAQCQBhCD8IGPM4AVg+om+ABVfb0rdTsFsAJJQPWwOZgNk1ACCBbmMCGRMp3wAvvWgA)
```sh
--html-whitespace-sensitivity strict
--parser html
--single-quote
```

**Input:**
```html
<div class="a class" i18n>lorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet.</div>
<div class="a class" i18n>lorem ipsum dolor sit ametlorem ipsum dolor </div>
```

**Output:**
```html
<div class="a class" i18n
  >lorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit
  ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet.</div
>
<div class="a class" i18n>lorem ipsum dolor sit ametlorem ipsum dolor </div>

```

**Expected behavior:**
```html
<div class="a class" i18n
  >lorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit  ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet.</div
>
<div class="a class" i18n>lorem ipsum dolor sit ametlorem ipsum dolor </div>

```

**Comment by thorn0** [comment]
> This is basically breaking the i18n translation, since it changes the hash of the string when generating messages.xlf

@voithos As the person who added support for `i18n` to Prettier's Angular formatter, can you confirm that this is indeed the case? Should tags with `i18n` be treated as `<pre>` by Prettier?

**Comment by voithos** [comment]
This is due to a long-standing bug in Angular that causes it to be whitespace-sensitive when generating i18n messages: https://github.com/angular/angular/issues/13114

I'd guess that treating `i18n` tags as `<pre>` wouldn't be ideal, especially for projects with heavy i18n requirements (in which case basically nothing would be formatted!). This is something that needs to be fixed in Angular itself. My understanding is that work on this has been stalled for ~years because of the new Ivy renderer, but since Ivy was recently released, it may now be possible to fix this.

**Comment by alexdinescu** [comment]
I don't think it would really help, all the discussions on the i18n angular issue is only related to leading/trailing spaces, which are already correctly handled in Prettier by `--html-whitespace-sensitivity strict`.
Even if the issue will get fixed in angular, I don't think it will include handling ( ignoring them when generating the id ) for whitespaces which are inside the translatable string. 
In our case, Prettier is adding line breaks inside the translatable string. The generated id would still get changed because of it even after a fix in angular.... if I didn't miss something

**Comment by voithos** [comment]
@alexdinescu, I think that's incorrect, the issue is talking about general whitespace sensitivity, not just leading/trailing spaces. [One of the examples](https://github.com/angular/angular/issues/13114#issuecomment-288251681) given in that issue is around "changing indentation", which is something that is likely to happen in the context of a formatter.

Your example of `green house` vs `greenhouse` doesn't apply because internal whitespace wouldn't  be _removed_, just _collapsed_ (browsers do this all the time). So `green house` wouldn't be the same thing as `greenhouse`, but it _would_ be the same thing as
```
     green      
         house
```

Which I think is exactly what we want, for i18n purposes.

**Comment by alexdinescu** [comment]
right, I missread the `replace(/\s+/, ' ')` ... 

meanwhile, it doesn't look that there is much / any progress on the angular issue... 
so maybe you can consider adding another property in the `--html-whitespace-sensitivity` that would just treat them as `<pre>` ? 
it would only be ..call it an workaround for this case, and existing users wouldn't be affected

**Comment by DanielRose** [comment]
Related to this issue:

The `i18n` tag can also be applied to attributes, via `i18n-<attributeName>`. For example:

**Input**

```
<div i18n-title title="lorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet."></div>
```

**Output**

```
<div
  i18n-title
  title="lorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet."
></div>
```

Prettier handles this correctly. However, if I include an Angular interpolation (and `angular` parser!), Prettier uses the interpolation to insert a line break, with the same problem of the generated hash now being different:

**Input with interpolation**

```
<div i18n-title title="lorem ipsum dolor sit ametlorem ipsum dolor sit {{ ametlorem }} ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet."></div>
```

**Output with interpolation**

```
<div
  i18n-title
  title="lorem ipsum dolor sit ametlorem ipsum dolor sit {{
    ametlorem
  }} ipsum dolor sit ametlorem ipsum dolor sit ametlorem ipsum dolor sit amet."
></div>
```
